books/kestrel/utilities/system/numbered-names.lisp

```diff
@@ -123,17 +126,25 @@
   "{")
 
 (define get-numbered-name-index-start ((wrld plist-worldp))
-  :returns (start "A @(tsee numbered-name-index-start-p).")
-  :verify-guards nil
+  :returns (start numbered-name-index-start-p "A @(tsee numbered-name-index-start-p).")
   :parents (numbered-name-index-start)
   :short "Retrieve the current starting marker
           of the numeric index of numbered names."
   :long
   "<p>
    If the starting marker is not set yet, the default is returned.
    </p>"
-  (let ((pair (assoc-eq 'start (table-alist 'numbered-name-index-start wrld))))
-    (if pair (cdr pair) *default-numbered-name-index-start*)))
+  (let ((pair (assoc-eq 'start (table-alist+ 'numbered-name-index-start wrld))))
+    (if (and pair
+             (numbered-name-index-start-p (cdr pair)))
```

Comment:
I think it’s fine as is, but in general I would throw an internal error if this condition fails, since (I think) we expect it to never fail. Something like
```
(raise “Internal error: ~x0 does not satisfy NUMBERED-NAME-INDEX-START-P.” (cdr pair))
```

Reply:
Isn't `(raise ...)` logically equivalent to `nil`? Presumably this would invalidate the returns lemma.

Reply:
Yes, great point. You can "adjust" the return type via
```
(prog2$ (raise ...) <anything of the right type>)
```
where `<anything of the right type>` will be never executed, but serves the logical purpose.

Reply:
Oh, of course, that makes perfect sense.